SimPEG/utils/model_utils.py

```diff
@@ -258,17 +263,45 @@ def distance_weighting(
 
     reference_locs = np.asarray(reference_locs)
 
-    # Calculate distance from receiver locations
-    # reference_locs is a scalar
-    if reference_locs.ndim < 2:
-        distance = np.abs(mesh.cell_centers[:, -1] - reference_locs)
+    cell_centers = mesh.cell_centers[active_cells]
+    cell_volumes = mesh.cell_volumes[active_cells]
+
+    # address 1D case
+    if mesh.dim == 1:
+        cell_centers = cell_centers.reshape(-1, 1)
+        reference_locs = reference_locs.reshape(-1, 1)
+
+    if engine == "numba":
+
+        @njit(parallel=True)
+        def distance_weighting_numba(
+            cell_centers: np.ndarray,
+            cell_volumes: np.ndarray,
+            reference_locs: np.ndarray,
+            threshold: float,
+            exponent: float = 2.0,
+        ):
+            distance_weights = np.zeros(len(cell_centers))
+            for _, rl in enumerate(reference_locs):
```

Comment:
No need to use `enumerate` if you're not assigning the index. Also, this for loop is not being run in parallel. To do so you need to use `numba.prange`: https://numba.pydata.org/numba-doc/latest/user/parallel.html

Something like:
```python
n_reference_locs = reference_locs.shape[0]  # check this, not sure if it's right
for i in prange(n_reference_locs):
    rl = reference_locs[i]
    ...
```